WooCommerce/Classes/ViewRelated/ReusableViews/SwiftUI Components/ReportListView.swift

```diff
@@ -0,0 +1,67 @@
+import SwiftUI
+import Yosemite
+
+protocol ReportListPresentable {
+    var items: [ReportItem] { get }
+    var title: String { get }
+    var ctaTitle: String { get }
+    var onDismiss: () -> Void { get }
+}
+
+struct ReportItem: Identifiable {
+    let id = UUID()
+    let title: String
+    let subtitle: String
+    let iconUrl: String
+    let iconBase64: String?
+
+    var icon: UIImage? {
+        guard let base64String = iconBase64 else { return nil }
+        return Data(base64Encoded: base64String).flatMap { UIImage(data: $0) }
+    }
+}
+
+/// Represent a screen with a list of IconListItems. Mainly used to present reports such as What's New in WooCommerce.
+///
+struct ReportListView: View {
+    let viewModel: ReportListPresentable
+    private var isPad: Bool { UIDevice.isPad() }
+
+    var body: some View {
+        VStack {
+            Spacer(minLength: isPad ? 40 : 75)
+            LargeTitleView(text: viewModel.title)
+            Spacer(minLength: isPad ? 32 : 40)
+            ScrollView {
+                ForEach(viewModel.items, id: \.id) {
+                    IconListItem(title: $0.title,
+                                 subtitle: $0.subtitle,
+                                 iconUrl: $0.iconUrl,
+                                 icon: $0.icon)
+                }
+            }
+            Spacer()
+            Button(viewModel.ctaTitle, action: viewModel.onDismiss)
+                .buttonStyle(PrimaryButtonStyle())
+                .padding(.horizontal, isPad ? 40 : 24)
+                .padding(.bottom, isPad ? 40 : 60)
+        }
+    }
+}
+
+// MARK: - Preview
+struct ReportListView_Previews: PreviewProvider {
+    static var features: [Feature] {
+        let jsonData = try? JSONSerialization.data(withJSONObject: ["title": "foo",
+                                                                    "subtitle": "bar",
+                                                                    "iconBase64": "",
+                                                                    "iconUrl": "https://s0.wordpress.com/i/store/mobile/plans-premium.png"],
+                                                   options: .fragmentsAllowed)
+        let feature = try? JSONDecoder().decode(Feature.self, from: jsonData ?? Data())
+        return [feature, feature, feature].compactMap { $0 }
```

Comment:
I'm not sure we are able to create a custom init for this model since it is Codable by default. This removed the ability to have memberwise inits. Another way around would be doing a bump on WordPressKit to make this change:

```swift
struct MyStruct {
    var name: String
}
extension MyStruct: Codable {} // preserves memberwise initializer

MyStruct(name: "Carlos")
```
Thoughts?

Reply:
The other, and probably better option now that I think about it, is to provide a memberwise initializer for your view model.

That way you can `init` it with that is needed.

```swift
init(items: [ReportItem], onDismiss: @escaping () -> Void) {
    self.items = items
    self.onDismiss = onDismiss
}
```

Then you can create it with `ReportItem` instances without having to mess with Json decoding in previews.

Reply:
Yep! Sounds about right :+1:

Reply:
Done! ✅